File: Java/Tema4-Exceptions/exceptions/ejercicio2/TestGato.java
package exceptions.ejercicio2;

/**
 * Test para probar la clase Gato.
 * 
 * @author Rafael Miguel Cruz Álvarez
 * @version 1.0
 */

public class TestGato {
  public static void main(String[] args) {
    
    Gato garfield = new Gato("macho");
    Gato tom = new Gato("macho");
    Gato titi = new Gato("hembra");
    try {
      System.out.println(garfield.apareaCon(titi));
      garfield.apareaCon(tom);
    } catch (ApareamientoNoPosibleException e) {
      System.err.println(e.getMessage());
    }
    
  }
}

File: Python/Exceptions/exceptions/gato/TestGato.py
'''
Programa para probar ejercicio gato con excepciones.
@author: Rafael Miguel Cruz Álvares
'''

from exceptions.gato import Gato

garfield = Gato("macho")
tom = Gato("macho")
lisa = Gato("hembra")

try:
    garfield.apareaCon(lisa)
except:
    print("Los gatos no pueden aparearse entre machos")